src/Prelude/Bool/Properties.agda

```diff
@@ -6,14 +6,52 @@ open import Prelude.Bool
 open import Prelude.Decidable
 open import Prelude.Empty
 
+de-morg-neg-conj : (a : Bool) → (b : Bool) → not (a && b) ≡ (not a || not b)
+de-morg-neg-conj true true = refl
+de-morg-neg-conj false true = refl
+de-morg-neg-conj true false  = refl
+de-morg-neg-conj false false  = refl
+
+de-morg-neg-disj : (a : Bool) → (b : Bool) → not (a || b) ≡ (not a && not b)
+de-morg-neg-disj true true = refl
+de-morg-neg-disj false true = refl
+de-morg-neg-disj true false  = refl
+de-morg-neg-disj false false  = refl
+
+true≢false : _≢_ {A = Bool} true false
+true≢false ()
+
+false≢true : _≢_ {A = Bool} false true
+false≢true ()
+
+x||true : {x : Bool} → (x || true) ≡ true
+x||true {true} = refl
+x||true {false} = refl
+
+x&&false : {x : Bool} → (x && false) ≡ false
+x&&false {true} = refl
+x&&false {false} = refl
+
+module _ {x : Bool} where
+
+  IsTrue⇒≡ : IsTrue x → x ≡ true
+  IsTrue⇒≡ true = refl
+
+  IsFalse⇒≡ : IsFalse x → x ≡ false
+  IsFalse⇒≡ false = refl
+
+  not[x]≡false⇒x≡true : {x : Bool} → not x ≡ false → x ≡ true
+  not[x]≡false⇒x≡true {true} _ = refl
+  not[x]≡false⇒x≡true {false} ()
+
+
+  not[x]≡true⇒x≡false : {x : Bool} → not x ≡ true → x ≡ false
+  not[x]≡true⇒x≡false {true} ()
+  not[x]≡true⇒x≡false {false} _ = refl
```

Comment:
Shadowing `x`. Nicer might be to import `Prelude.Variables` and drop the anonymous module.

Reply:
Oh yeah, I should have moved those out of the anonymous module. I have a strange issue with the definition here here the removal of **`{x}`******

Reply:
I had a similar problem with the two definitions `x||true` and `x&&false`. I'm not quite sure, maybe you know?

Reply:
Yes, you cannot bind the module parameter in the left-hand side, since it's already bound in the module telescope. Importing `Prelude.Variables` will solve this and let you write
```agda
  not[x]≡true⇒x≡false : not x ≡ true → x ≡ false
  not[x]≡true⇒x≡false {true} ()
  not[x]≡true⇒x≡false {false} _ = refl
```